src/js/components/ServiceDetailDebugTab.js

```diff
@@ -0,0 +1,114 @@
+import moment from 'moment';
+import React from 'react';
+
+import DescriptionList from './DescriptionList';
+import Service from '../structs/Service';
+import TaskStatsTable from './TaskStatsTable';
+
+class ServiceDetailDebugTab extends React.Component {
+  getValueText(value) {
+    if (value == null || value === '') {
+      return (
+        <p>Unspecified</p>
+      );
+    }
+
+    return (
+      <span>{value}</span>
+    );
+  }
+
+  getLastTaskFailureInfo() {
+    let {lastTaskFailure} = this.props.service;
+    if (lastTaskFailure == null) {
+      return (
+        <p>This app does not have failed tasks</p>
+      );
+    }
+
+    const {version, timestamp, taskId, state, message, host} = lastTaskFailure;
+    let timeStampText = 'Just now';
+    if (new Date(timestamp) > Date.now()) {
+      timeStampText = new moment(timestamp).fromNow();
+    }
+    let taskFailureValueMapping = {
+      'Task id': this.getValueText(taskId),
+      'State': this.getValueText(state),
+      'Message': this.getValueText(message),
+      'Host': this.getValueText(host),
+      'Timestamp': <span>{timestamp} ({timeStampText})</span>,
+      'Version': <span>{version} ({new moment(version).fromNow()})</span>
+    };
+
+    return (<DescriptionList hash={taskFailureValueMapping} />);
+  }
+
+  getLastVersionChange() {
+    let {versionInfo} = this.props.service;
+    if (versionInfo == null) {
+      return (
+        <p>This app does not have version change information</p>
+      );
+    }
+
+    const {lastScalingAt, lastConfigChangeAt} = versionInfo;
+    let lastScaling = 'No operation since last config change'
+    if (lastScalingAt !== lastConfigChangeAt) {
+      lastScaling = (
+        <span>{lastScalingAt} ({new moment(lastScalingAt).fromNow()})</span>
+      );
+    }
+
+    let LastVersionChangeValueMapping = {
+      'Scale or Restart': lastScaling,
+      'Configuration': (
+        <span>
+          {lastConfigChangeAt} ({new moment(lastConfigChangeAt).fromNow()})
+        </span>
+      )
+    };
+
+    return (<DescriptionList hash={LastVersionChangeValueMapping} />);
+  }
+
+  getTaskStats() {
+    let {taskStats} = this.props.service;
```

Comment:
Why aren't you using the service getter? Using [`Service.prototype. getTaskStats`](https://github.com/dcos/dcos-ui/blob/master/src/js/structs/Service.js#L168) would give you `TaskStats` struct which would save you a lot of unnecessary fiddling with the raw data.